docs/configure-artifact-repository.md

```diff
@@ -107,43 +110,176 @@ $ cat > policy.json <<EOF
    ]
 }
 EOF
-$ aws s3 mb s3://$mybucket [--region xxx]
-$ aws iam create-user --user-name $mybucket-user
-$ aws iam put-user-policy --user-name $mybucket-user --policy-name $mybucket-policy --policy-document file://policy.json
-$ aws iam create-access-key --user-name $mybucket-user > access-key.json
 ```
 
 If you do not have Artifact Garbage Collection configured, you should remove `s3:DeleteObject` from the list of Actions above.
 
-NOTE: if you want argo to figure out which region your buckets belong in, you
-must additionally set the following statement policy. Otherwise, you must
-specify a bucket region in your workflow configuration.
+!!! Note "Discovering bucket regions"
+    If you want Argo Workflows to figure out which region your buckets belong in, you must additionally set the following statement policy.
+    Otherwise, you must specify a bucket region in your workflow configuration.
 
 ```json
       {
-         "Effect":"Allow",
-         "Action":[
+        "Effect":"Allow",
+        "Action":[
             "s3:GetBucketLocation"
-         ],
-         "Resource":"arn:aws:s3:::*"
+        ],
+        "Resource":"arn:aws:s3:::*"
       }
     ...
 ```
 
 ### AWS S3 IRSA
 
-If you wish to use S3 IRSA instead of passing in an `accessKey` and `secretKey`, you need to annotate the service account of both the running workflow (in order to save logs/artifacts) and the argo-server pod (in order to retrieve the logs/artifacts).
+Using IRSA is the recommended Kubernetes' native mechanism.
+If you are using AWS EKS, see [the AWS documentation to enable it in your cluster](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html).
+If you are not, see the [EKS Pod Identity Webhook instructions](https://github.com/aws/amazon-eks-pod-identity-webhook/blob/master/SELF_HOSTED_SETUP.md).
```

Comment:
```suggestion
IAM Roles for Service Accounts (IRSA) is the recommended Kubernetes native mechanism to authenticate to S3.
If you are using EKS, follow [the IRSA setup guide](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html).
If not, follow the [Pod Identity Webhook self-hosted setup guide](https://github.com/aws/amazon-eks-pod-identity-webhook/blob/master/SELF_HOSTED_SETUP.md).
```

- Use the full name somewhere
- "mechanism to authenticate with S3", answering the question "mechanism for what?"
- consistently use the wording "follow the X setup guide" for the two options
  - remove the "EKS" portion from the second one to confuse users less
  - (also last I checked, the Pod Identity Webhook failed on self-hosted clusters and was still pretty alpha)

Reply:
Sounds good. I'm actually not sure about the webhook on self-hosted clusters - we are using EKS. I included this for posterity, since the docs previously made no mention about how IRSA is used in either case. If you're concerned about it still being pretty experimental, I can remove this line entirely to not mislead users.

Reply:
I've been using kOps for ~7 years; last I tried the EKS webhook was 2+ years ago and it did not work -- I don't remember why but I remember it was an unresolved issue and there are more of those in the repo now 😕

The (less optimal) alternatives that pre-date IRSA are [`kiam`](https://github.com/uswitch/kiam) and [`kube2iam`](https://github.com/jtblin/kube2iam) and perhaps others. I was thinking we perhaps link to a few, but I see that `kiam` was abandoned in the past year https://github.com/uswitch/kiam/pull/528 and `kube2iam` is similarly lacking updates https://github.com/jtblin/kube2iam/issues/376 https://github.com/jtblin/kube2iam/issues/373.

So 😕
 I'm fine leaving the self-hosted guide on as it did not get much attention in the past and perhaps more attention will mean AWS maintains it better(??)